### ISSUE TITLE
Struct Initialization: item_brace_style AlwaysNextLine

### ISSUE THREAD

**Issue by rickvanprim** [rust-lang/rustfmt#2105]
With `item_brace_style = "AlwaysNextLine"`:

Expected
```
let foo = Foo
{
  a = 1,
  b = 2,
};
```

Actual
```
let foo = Foo {
  a = 1,
  b = 2,
};
```

**Comment by topecongiro** [comment]
Should `item_brace_style` change the brace position of struct literal? It is an expression, not an item (at least in theory).

**Comment by nrc** [comment]
> It is an expression, not an item

Good point! I'm not really sure. Probably not, since items and expressions are sometimes formatted differently. OTOH, I don't really want to add another option for this and I think a person who likes the 'brace on its own line' style might expect both items and expressions to be formatted the same way.

**Comment by rickvanprim** [comment]
Any new thoughts on this?

**Comment by ytmimi** [comment]
@calebcartwright is this a bug or intended behavior? Also `item_brace_style` no longer exists. To simulate the original issue we can set `brace_style=AlwaysNextLine` and `struct_lit_single_line=false`

**input**
```rust
fn main() {
    let foo = Foo {
        a: 1,
        b: 2,
    };
}
```

**output**
```rust
fn main()
{
    let foo = Foo {
        a: 1,
        b: 2,
    };
}
```

**Comment by ytmimi** [comment]
Linking tracking issue for `brace_style` (#3376). Although `struct_lit_single_line` (#3357) doesn't cause any issues i'm go to link it just in case since I feel like we'd need to figure out what to do about the interaction between `brace_style=AlwaysNextLine` and `struct_lit_single_line=true`.
